typescript/sdk/src/HyperlaneApp.ts

```diff
@@ -26,39 +26,39 @@ export class HyperlaneApp<
     super(connectedContractsMap);
   }
 
-  static buildContracts<C extends HyperlaneContracts>(
-    addresses: ChainMap<HyperlaneAddresses>,
-    factories: HyperlaneFactories,
+  static buildContracts<F extends HyperlaneFactories>(
+    addresses: HyperlaneAddressesMap<any>,
+    factories: F,
     multiProvider: MultiProvider,
-  ): { contracts: ChainMap<C>; intersectionProvider: MultiProvider } {
-    const chains = Object.keys(addresses);
+  ): {
+    contracts: HyperlaneContractsMap<F>;
+    intersectionProvider: MultiProvider;
+  } {
+    const filteredAddresses = objFilter(
+      addresses,
+      (chain, addrs): addrs is HyperlaneAddresses<F> => {
+        return Object.keys(factories).every((contract) =>
+          Object.keys(addrs).includes(contract),
+        );
+      },
```

Comment:
not totally clear why we need this independent of the other filterAddresses?

Reply:
One of them filters out addresses that are not in our factories, the other filters out chains that do not have a complete set of addresses.

I've renamed some things to hopefully make it clearer